### PR TITLE
HIVE-16893 : move replication dump related work in semantic analysis phase to execution phase using a task

### PR DESCRIPTION
**Pull request by anishek**

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java [hunk 1]
```diff
@@ -299,18 +184,6 @@ private void analyzeReplDump(ASTNode ast) throws SemanticException {
...
-        db,
-        conf,
-        getNewEventOnlyReplicationSpec(ev.getEventId())
-    );
-    EventHandlerFactory.handlerFor(ev).handle(context);
-    REPL_STATE_LOG.info("Repl Dump: Dumped event with ID: {}, Type: {} and dumped metadata and data to path {}",
-                        String.valueOf(ev.getEventId()), ev.getEventType(), evRoot.toUri().toString());
-  }
 
   public static void injectNextDumpDirForTest(String dumpdir){
```
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java]: Remove the unused methods such as removeValuesTemporaryTables, matchesTbl, matchesDb, getNewEventOnlyReplicationSpec, getNewReplicationSpec from ReplicationSemanticAnalyzer.java.

Also, update removeTemporaryTablesForMetadataDump test method to refer ReplDumpWork class.

File: ql/src/java/org/apache/hadoop/hive/ql/exec/TaskFactory.java [hunk 2]
```diff
@@ -108,7 +110,7 @@ public TaskTuple(Class<T> workClass, Class<? extends Task<T>> taskClass) {
         IndexMetadataChangeTask.class));
     taskvec.add(new TaskTuple<TezWork>(TezWork.class, TezTask.class));
     taskvec.add(new TaskTuple<SparkWork>(SparkWork.class, SparkTask.class));
-
+    taskvec.add(new TaskTuple<>(ReplDumpWork.class, ReplDumpTask.class));
```
Comment (anishek) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/TaskFactory.java]: it picks up the corresponding class from the arguments, with java 1.8 specifying the above is redundant.

File: ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java (hunk 1 above)
Comment (anishek) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java]: done

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+    HiveWrapper.Tuple<Database> database = new HiveWrapper(getHive(), dbName).database();
+    EximUtil.createDbExportDump(fs, dumpPath, database.object, database.replicationSpec);
+    REPL_STATE_LOG.info("Repl Dump: Dumped DB metadata");
+    return dbRoot;
+  }
+
+  private void dumpTable(String dbName, String tblName, Path dbRoot) throws Exception {
+    try {
+      Hive db = getHive();
+      BaseSemanticAnalyzer.TableSpec
```
Comment (anishek) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: done

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+        new DatabaseAndTableFilter(work.dbNameOrPattern, work.tableNameOrPattern);
+    EventUtils.MSClientNotificationFetcher evFetcher =
+        new EventUtils.MSClientNotificationFetcher(getHive().getMSC());
+    EventUtils.NotificationEventIterator evIter = new EventUtils.NotificationEventIterator(
+        evFetcher, bootDumpBeginReplId,
+        Ints.checkedCast(bootDumpEndReplId - bootDumpBeginReplId) + 1,
+        evFilter);
+
+    // Now we consolidate all the events that happenned during the objdump into the objdump
+    while (evIter.hasNext()) {
```
Comment (anishek) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: yeh i was not sure when it was required so i have left it for now to be a gentle reminder, may be we discuss this in next catch up and create a separate task for this.

File: ql/src/java/org/apache/hadoop/hive/ql/exec/TaskFactory.java (hunk 2 above)
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/TaskFactory.java]: ok

File: ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java (hunk 4 above)
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java]: ok

File: ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java [hunk 5]
```diff
@@ -19,40 +19,23 @@ Licensed to the Apache Software Foundation (ASF) under one
 
 import com.google.common.base.Predicate;
 import com.google.common.collect.Collections2;
-import com.google.common.primitives.Ints;
 import org.antlr.runtime.tree.Tree;
 import org.apache.hadoop.fs.FileStatus;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.hive.conf.HiveConf;
```
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java]: Need to remove import of class HiveConf which is unused.

**Pull request closed without merging**